### PR TITLE
ERM-508: Support multiple POLs on a single Agreement Line

### PR DESCRIPTION
**Pull request by CalamityC**
* change Entitlement domain class
* add OrderLine, OrderLineController, _orderLine.gson
* update URLMappings
* update database migrations files

### CODE REVIEW

**Comment by doytch** [comment]
This is failing for me with agreements that existed prior to switching onto the branch.
```
2019-11-05 18:54:32.648 ERROR --- [  XNIO-2 task-4] StackTrace                               : Full Stack Trace:

org.postgresql.util.PSQLException: ERROR: relation "order_line" does not exist
  Position: 254
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
```
Migration issue?

File: service/grails-app/domain/org/olf/erm/OrderLine.groovy [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package org.olf.erm;
+
+import com.k_int.okapi.remote_resources.RemoteOkapiLink
+import com.k_int.web.toolkit.refdata.Defaults
+import com.k_int.web.toolkit.refdata.RefdataValue
+
+import grails.gorm.MultiTenant
+
+public class OrderLine extends RemoteOkapiLink implements MultiTenant<OrderLine> {
```
[the remark this answers is not in the document]
Comment (sosguthorpe) [reply service/grails-app/domain/org/olf/erm/OrderLine.groovy]: Use something like Period instead. Remove the fields you don't need. Change the owner and add the poLineId field.

[the remark this answers is not in the document]
Comment (CalamityC) [reply service/grails-app/domain/org/olf/erm/OrderLine.groovy]: I think it's ok as it is now.
Anything else to change?

[the remark this answers is not in the document]
Comment (sosguthorpe) [reply service/grails-app/domain/org/olf/erm/OrderLine.groovy]: Yes. It still says `extends RemoteOkapiLink` which it shouldn't now. Remote Okapi link has special functionality attached to it which we are removing from OrderLine for now. So:
`public class OrderLine extends RemoteOkapiLink implements MultiTenant<OrderLine>`
becomes...
`public class OrderLine implements MultiTenant<OrderLine>`

After you have removed that you will need to add the fields that were inherited from that object. Like the `String id` and `String poLineId`.

File: service/grails-app/migrations/update-mod-agreements-1-7.groovy [hunk 2]
```diff
@@ -1,9 +1,4 @@
 databaseChangeLog = {
-  changeSet(author: "sosguthorpe (generated)", id: "1558523001530-1") {
-    addColumn(tableName: "entitlement") {
-      column(name: "ent_po_line_id", type: "varchar(255)")
-    }
-  }
```
Comment (sosguthorpe) [inline service/grails-app/migrations/update-mod-agreements-1-7.groovy]: You should **never** delete from existing migrations.Please just extend your migrations to remove the column (`dropColumn`) like: https://github.com/folio-org/mod-agreements/blob/master/service/grails-app/migrations/update-mod-agreements-2-0.groovy#L59

Comment (CalamityC) [reply service/grails-app/migrations/update-mod-agreements-1-7.groovy]: Fixed it. Thanks for the link, didn't know the dropColumn syntax yet

Comment (sosguthorpe) [reply service/grails-app/migrations/update-mod-agreements-1-7.groovy]: No problem.

File: service/grails-app/migrations/order-line.groovy [hunk 3]
```diff
@@ -0,0 +1,30 @@
+databaseChangeLog = {	
+	changeSet(author: "claudia (manual)", id: "2019-11-04-00001") {
+		createTable(tableName: "order_line") {
+			column(name: "pol_id", type: "VARCHAR(36)") {
+				constraints(nullable: "false")
+			}
+			column(name: "pol_version", type: "BIGINT") {
+				constraints(nullable: "false")
+			}
+			column(name: "pol_orders_fk", type: "VARCHAR(50)") {
```
[the remark this answers is not in the document]
Comment (CalamityC) [reply service/grails-app/migrations/order-line.groovy]: this is supposed to be the field where the poLineId from orders is stored

**Comment by sosguthorpe** [comment]
This all looks really great @CalamityC . I will hold off on merging as I just need to test the migrations locally. Great job!

**Pull request merged**